### ISSUE TITLE
delayed_job consumes one core

### ISSUE THREAD

**Issue by haslinger** [collectiveidea/delayed_job#880]
I see delayed job consuming one core. Last think I did was updating my app to rails 4.2.5.
So I now also updated delayed job (to 4.1.1), but that didn't change a thing.

I entered ```config.active_job.queue_adapter = :delayed_job``` in application.rb as stated in the readme.

I am attaching a load diagram, because I think, it's an interesting pattern.

![bildschirmfoto](https://cloud.githubusercontent.com/assets/1116624/13005582/22aedb62-d183-11e5-806c-1e8b2f58d29e.png)

Needless to mention, that there is no current job in the queue.
All the load comes from delayed job and stopping it means killing it:

```
stefan@aristoteles:/var/rails/mercator-update$ bin/delayed_job stop
/home/stefan/.rvm/gems/ruby-2.1.5/gems/mime-types-2.99/lib/mime/types.rb:274:in `index_extensions': MIME::Types#index_extensions is deprecated and will be private.
delayed_job: trying to stop process with pid 14542...
delayed_job: process with pid 14542 won't stop, we forcefully kill it...
delayed_job: process with pid 14542 successfully stopped.
```

Now I am mostly searching for a way to find out, what's going on, as I don't see a log about the activity. Any advice in any way appreciated.

Update: The nice think about this issue is, that it only happens on my development system (laptop), in the production system, everything is fine. Pheew!

**Comment by albus522** [comment]
Since it had to be force killed, my guess is that something got caught in an infinite loop before DJ could initialize. If DJ had initialized, it would have been in the worker loop and would have exited quickly as long as a job wasn't running. Given you are getting startup warnings, there are probably other incompatible gems in your gemset. I would start by getting everything as up to date as you can

**Comment by haslinger** [comment]
And that deprecation warning would come over and over again, when I stay on top with ```delayed_job run```. Even more interesting, the jobs are still handled properly. Digging deeper ...

**Comment by haslinger** [comment]
OK, I got it. It seems to be the code reloading feature. I'm using a gem called hobo, that precompiles views which takes some time ... And every roughly 10 secs delayed job seems to reload the app. 

There is a another gem rails-dev-boost which speeds up development mode in rails a lot and enabling it for my app reduced the load to below 0.2 which indicated to me, that there is an issue with code reloading here.
Well, so I went the next step and configured my development mode to ```config.cache_classes = false``` and hurray, after the app has been fully loaded, there is no load on the processor any more .

In other terms, it would be nice to have a config option to disable the auto load in development mode. 
Or I shouldn't run delayed job in development mode at all. :-)
I guess this issue can be closed then.
